virttest/utils_libvirt/libvirt_disk.py

```diff
@@ -24,6 +24,83 @@
 LOG = logging.getLogger('avocado.' + __name__)
 
 
+def get_non_root_disk_name(session):
+    """
+    Returns the first disk name under /dev whose device doesn't have any
+    partitions or volumes or itself mounting "/".
+    Assuming that the system is installed on a single disk, this
+    can be used to identify an added disk.
+    It also includes read-only devices as such, e.g. sr0.
+
+    It iterates over output with the following structure in mind:
+    vda
+    ├─vda1        /boot
+    └─vda2
+      ├─rhel-root /
+      └─rhel-swap [SWAP]
+    But also considers corner cases where everything is for example on vda and
+    no swap.
+
+    :param session: If given the command will be executed in this VM or
+                    remote session.
+    :return (name, mount_point): A tuple containing the device name and the mount
+                    point. If the information doesn't exist it's returned as empty
+                    string.
+    :raises TestError: If there's no additional disk or no mount on "/" or when
+                       the disk information cannot be retrieved.
+    """
+    cmd = "lsblk -n -o NAME,MOUNTPOINT"
+    s, o = utils_misc.cmd_status_output(cmd,
+                                        ignore_status=False,
+                                        shell=True,
+                                        verbose=True,
+                                        session=session)
+    if s:
+        raise exceptions.TestError("Couldn't list block devices: '%s'" % o)
+    LOG.debug("lsblk output:\n%s", o)
+    lines = o.split("\n").copy()
+    root_disk = None
+    root_mounted = False
+    disk_pattern = re.compile(r'^([a-z]+|sr\d)[\s\t$]')
+    entry_pattern = re.compile(r'(.*)[\s\t]+(.*)')
+    idx = 0
+    while -1 < idx < len(lines) + 1:
+        line = lines[idx]
+        entry = entry_pattern.match(line)
+        if not entry:
+            idx = idx + 1
+            continue
+        name, mpoint = entry.groups()
+        is_disk = disk_pattern.match(name)
+        if root_mounted:
+            if is_disk:
+                root_disk = line
+                break
+            else:
+                idx = idx - 1
+                continue
```

Comment:
Just confirm, We will never run into this `if root_mounted` branch if we set  ` root_mounted = False` in L63 and not pass it in `def get_non_root_disk_name(session)` right?

Reply:
@nanli1 That is correct this branch will not be entered if root_mounted is never set to True in L. 83. This means there is no mount point for "/" (L. 82), therefore the idx will just go up until len(lines) and the TestError is raised  (L. 94).